### PR TITLE
Move to bluebird

### PR DESCRIPTION
**Pull request by mustafaiman**
I changed from Q to Bluebird. It should not affect end users as we still return Promises.
I advise doing a clean install at your local after this PR is merged:
```
npm run clean
rm -r node_modules
npm install
npm run compile
```

### CODE REVIEW

**Comment by mustafaiman** [comment]
Unrelated test failure: https://github.com/hazelcast/hazelcast-nodejs-client/issues/124

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
@GhostInAMachine review please

File: src/HazelcastClient.ts [hunk 1]
```diff
@@ -61,8 +61,8 @@ export default class HazelcastClient {
         this.heartbeat = new Heartbeat(this);
     }
 
-    private init(): Q.Promise<HazelcastClient> {
-        var deferred = Q.defer<HazelcastClient>();
+    private init(): Promise<HazelcastClient> {
+        var deferred = Promise.defer<HazelcastClient>();
```
Comment (GhostInAMachine) [inline src/HazelcastClient.ts]: This is deprecated. 
http://bluebirdjs.com/docs/api/deferred-migration.html

**Comment by mustafaiman** [comment]
I currently removed some deferreds but not all of them. I'll change all of them but I would like to tell my concerns about removing deferreds;
- Deferreds suits our invocation system best. We store deferreds in `pendingQueue` and resolve when response to that request is received. To do that without using deferred objects, we need to store specific `resolve` and `reject` functions to each invocation. This seems a little odd. In fact, Bluebird's migration quide code simulates the same phenomenon using Bluebird's API but not native support. So they even could not find a proper better way to replace what they call `discouraged`.
- Promise constructor pattern is noticeably slower compared to deferreds. See 4th paragraph of Esailija's answer here: http://programmers.stackexchange.com/questions/278778/why-are-native-es6-promises-slower-and-more-memory-intensive-than-bluebird. Esailjia is the author of Bluebird and he himself says Promise constructor pattern is slow by definition. Then Bluebird deprecates deferred pattern. This does not make sense. Keep in mind that Bluebird is claiming to be faster than native promises(they really are) and they explain why they are faster by commenting on inherently slow constructor of native API.
- Another Bluebird author says deferreds are likely to be never removed here: https://github.com/petkaantonov/bluebird/issues/555#issuecomment-88588920 . Also the reason he gives to justify the deprecation does not apply to our case in Invocations.
- Bluebird documentation says **Use deferreds when you have to**. See `When should deferred be used` section in https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns. So they admit deferreds need to be present in the library. So they are likely to be never removed.

On the other hand, I think we should not count on library and use only Promises/A+ specification. I will change deferreds to Promise constructor pattern but note that this will have some performance drawback.

**Pull request merged**